src/main/java/org/colonelkai/publictransit/line/Line.java

```diff
@@ -40,34 +43,54 @@ private Line(String identifier, String name, List<Node> nodes, double cost, Cost
         this.oneWay = oneWay;
         this.oneWayReversed = oneWayReversed;
         this.nodes = new ArrayList<>(nodes);
-        this.validateNodes();
+        this.validate(this.nodes);
     }
 
     Line(@NotNull LineBuilder builder) {
         this.identifier = Objects.requireNonNull(builder.identifier());
-        this.name = Objects.requireNonNullElse(builder.name(), this.identifier);
+        this.name = Objects.requireNonNullElse((null == builder.name()) ? null : builder.name().toPlain(), this.identifier); //this needs fixing
         this.nodes = builder.nodes().stream().map(NodeBuilder::build).toList();
-        this.validateNodes();
         this.cost = Objects.requireNonNull(builder.cost());
         this.costType = Objects.requireNonNull(builder.costType());
         this.oneWay = builder.isOneWay();
         this.oneWayReversed = builder.isOneWayReversed();
+        this.validate(this.nodes);
     }
 
-    private void validateNodes() {
-        if (2 > this.nodes.size()) {
-            throw new IllegalStateException("Requires two or more nodes");
-        }
-        Collection<Node> uniqueTest = new HashSet<>(this.nodes);
-        if (uniqueTest.size() != this.nodes.size()) {
-            throw new IllegalStateException("Two or more nodes have the same name");
+    private void validate(Collection<Node> nodes) {
+        Collection<Node> uniqueTest = new HashSet<>(nodes);
+        if (uniqueTest.size() != nodes.size()) {
+            throw new IllegalStateException("Two or more nodes have the same position");
         }
     }
 
+    public boolean isActive() {
+        return 2 <= this.nodes.stream().filter(node -> NodeType.STOP == node.getNodeType()).count();
+    }
+
     public List<Node> getNodesBetween(Node start, Node end) {
         int startIndex = this.nodes.indexOf(start);
         int endIndex = this.nodes.indexOf(end);
-        return this.getNodes().subList(startIndex, endIndex);
+        return this.applyReverseLogic(this.nodes.subList(Math.min(startIndex, endIndex), Math.max(startIndex, endIndex)), false);
+    }
+
+    public Optional<Node> getNextNode(Node node) {
+        return this.getNextNode(this.getNodes().indexOf(node));
+    }
+
+    public Optional<Node> getNextNode(int index) {
+        index++;
```

Comment:
"Next Node" may not necessarily be the next index, the player could be going backwards in the line, or the it could be one way only on the reverse direction. I hope I'M not misreading this.

Reply:
I just noticed `getNodes()` already calls `applyReverseLogic()`, so no worries my bad.